src/components/tl/nccl/tl_nccl_coll.c

```diff
@@ -419,14 +419,42 @@ ucc_status_t ucc_tl_nccl_bcast_start(ucc_coll_task_t *coll_task)
     ucc_status_t        status = UCC_OK;
     size_t              count  = args->src.info.count;
     ucc_rank_t          root   = args->root;
+    ucc_rank_t          peer, rank, size;
     ncclDataType_t      dt;
+    ucc_ep_map_t        map;
 
     dt = ucc_to_nccl_dtype[UCC_DT_PREDEFINED_ID(args->src.info.datatype)];
     task->super.status = UCC_INPROGRESS;
     UCC_TL_NCCL_PROFILE_REQUEST_EVENT(coll_task, "nccl_bcast_start", 0);
-    NCCLCHECK_GOTO(ncclBroadcast(src, src, count, dt, root, team->nccl_comm,
-                                 stream),
-                   exit_coll, status, UCC_TL_TEAM_LIB(team));
+
+    if (UCC_COLL_ARGS_ACTIVE_SET(args)) {
+        map  = ucc_active_set_to_ep_map(args);
```

Comment:
I guess this is fine for now since `size` will always be 2, but this is a a bottleneck in the general case.

Reply:
it is. This is what i meant during WG, when i talked about adding "my_ep (my_rank)" field as part of active set. I said, that w/o it i need O(active_set_size) loop to compute it) @manjugv fyi

Reply:
I'm not sure what the best solution is here. I think it's weird either way. It's not obvious what should go in this field to the user (you set it to `0` for both processes in your example), but it hurts performance without it...